### PR TITLE
Fix `ground` in `string`

### PR DESCRIPTION
**Pull request by samscott89**

### CODE REVIEW

File: polar-core/tests/integration_tests.rs [hunk 1]
```diff
@@ -1868,4 +1868,6 @@ fn test_builtin_iterables() {
         "x",
         vec![value!(1), value!(2)],
     );
+
+    qeval(&mut p, r#"forall(x in "abc", x in "abacus")"#);
```
Comment (gj) [inline polar-core/tests/integration_tests.rs:1872]: Does this only match b/c all of `"a"`, `"b"`, and `"c"` are in `"abacus"`? Could we add a complementary `qnull` test when that doesn't hold?

Comment (samscott89) [reply polar-core/tests/integration_tests.rs:1872]: Yep, will add a few more tests. Just wanted to push this before checkin so people could see the problem.

Comment (samscott89) [reply polar-core/tests/integration_tests.rs:1874]: @gj added

File: polar-core/src/vm.rs [hunk 2]
```diff
@@ -1485,7 +1485,7 @@ impl PolarVirtualMachine {
                             s.chars()
                                 .map(|c| c.to_string())
                                 .map(Value::String)
-                                .filter(|c| !is_ground || term.value() == c)
+                                .filter(|c| !is_ground || c == value)
```
Comment (gj) [inline polar-core/src/vm.rs:1488]: To ensure I have this correct: `term.value()` is the entire `in` operation, right? Now we're comparing `c` (a character produced by iterating over the string) and `value`, the LHS of the `in` operation?

**Review by gj — APPROVED** [review]

**Comment by gj** [comment]
Oh and please add a changelog entry for the bugfix.

File: polar-core/src/vm.rs (hunk 2 above)
Comment (samscott89) [reply polar-core/src/vm.rs:1488]: Yep! The previous behaviour made no sense, but it never got past the `!is_ground` check.

**Pull request merged**